xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -255,9 +255,11 @@ Object.extend(XWiki, {
                         }
                       });
 
-                      // Replace the element's content with the temporary container's content, while also evaluating any inline scripts.
-                      // Note: This also does stripScripts internally, but it's OK now, since we have already added the necessary external scripts to head.
-                      $(extraID + "pane").update(container.innerHTML);
+                      // jQuery's html() is used instead of prototype's update to avoid the undesired
+                      // interpretation of the innerHTML during the dom update.  
```

Comment:
> interpretation of the innerHTML

WDYM by this "interpretation"? What exactly is the difference between jQuery's ``html()`` and Prototype's ``update()``. Last time I checked we still had templates with in-line scripts so not executing in-line scripts is not an option.

Reply:
I've updated the approach and using the update method with the children of the container element instead of its innerHTML allows scripts to be actually executed where the content of the tags attributes are properly escaped.